src/libnml/os_intf/inetnull.cc

```diff
@@ -66,7 +70,11 @@ int inet_file_eof(INET_FILE * inet_file)
     if (NULL == inet_file) {
 	return 1;
     }
-    return feof(inet_file->fp);
+    if (inet_file->fp) {
+        return feof(inet_file->fp);
+    } else {
+	return 1; // or raise an error
```

Comment:
I checked the return value of feof() if given a bogus file pointer, and it should return -1 and set errno to EBADF.  Returning the same value as if inet_file is NULL seem to me to be the best way to handle bogus input, ie return 1 (and not raise an error).  A bogus file pointer can definitely be seen as at its end.  There is nothing more to read from it.

In short, I believe this patch should be ready to merge into master.